services/api.py

```diff
@@ -685,6 +687,15 @@ def get_queryset(self):
             queryset = queryset.prefetch_related('service_details__service')
 
         filters = self.request.query_params
+
+        # query parameter validation
+        f = {k: v[0].split(',') for k, v in filters.lists()}
+        try:
+            RequestFilters(**f)
+            UnitRequestFilters(**f)
+        except pydantic_ValidationError as e:
```

Comment:
See the import comment above. After changing that, I would just write

    except pydantic.ValidationError as e